Communication/src/com/gymfox/communication/Network.java

```diff
@@ -0,0 +1,188 @@
+        }
+
+        void setFirstUsableAddress(long address) {
+            firstUsableAddress = new IPv4Address(address + 1);
+        }
+
+        void setLastUsableAddress() {
+            lastUsableAddress = new IPv4Address(broadcastAddress.getIpLong() - 1);
+        }
+
+        void setMaskAsString(long address) {
+            long[] octets = new long[4];
+
+            octets[0] = ( address & 0xFF000000L ) >> 24;
+            octets[1] = ( address & 0x00FF0000L ) >> 16;
+            octets[2] = ( address & 0x0000FF00L ) >> 8;
+            octets[3] = ( address & 0x000000FFL ) ;
+
+            maskAsString = String.format("%d.%d.%d.%d", octets[0], octets[1], octets[2], octets[3]);
+        }
+
+        void setTotalHosts() {
+            totalHosts = ((1 << (MAX_MASK_VALUE - maskAsInt)) - RESERVED_ADDRESS);
+        }
+
+        void setPublic(Network address) {
+            for ( IPv4Address pa : privateAddress ) {
+                if ( address.getNetworkAddress().equals(pa) ) {
+                    isPublic = false;
+                }
+            }
+        }
+    }
+
+    public Network(IPv4Address address) {
+        this(address, DEFAULT_MASK);
+    }
+
+    public Network(IPv4Address address, int maskLength) {
+        this.maskAsLong = maskToLong(maskLength);
+        this.maskLength = maskLength;
+        this.address = new IPv4Address(address.getIpLong() & getMaskAsLong());
+        this.networkMetadata = new NetworkMetadata(this);
+        this.networkAsString = networkToString();
+    }
+
+    private long maskToLong(int maskLength) {
+        validate(maskLength);
+        int shift = MAX_MASK_VALUE - maskLength;
+
+        return LONG_MAX_IP << shift;
+    }
+
+    private void validate(int maskLength) {
+        if ( maskLength < MIN_MASK_VALUE || maskLength > MAX_MASK_VALUE ) {
+            throw new InvalidMaskValueException("Invalid Mask");
+        }
+    }
+
+    public boolean contains(IPv4Address address) {
+        return address.greaterThan(getNetworkAddress()) && address.lessThan(getBroadcastAddress());
+    }
+
+    private String networkToString() {
+        StringBuffer out = new StringBuffer();
+
+        out.append("Network: " + getNetwork() + "\n");
+        out.append("Network address: " + getNetworkAddress().getIpString() + "\n");
+        out.append("Network broadcast: " + getBroadcastAddress().getIpString() + "\n");
+        out.append("First address: " + getFirstUsableAddress().getIpString() + "\n");
+        out.append("Last address: " + getLastUsableAddress().getIpString() + "\n");
+        out.append("Total hosts: " + getTotalHosts() + "\n");
+        out.append("Is public: " + isPublic() + "\n");
+
+        return out.toString();
+    }
+
+    @Override
+    public String toString() {
+        return networkAsString;
+    }
+
+    public String getNetwork() {
+        return getNetworkAddress().getIpString() + "/" + getMaskLength();
```

Comment:
Looks like you always will get the same string. Probably can keep it as a field. 
Also method name a bit confusing. You're getting network, but returning string